tls/s2n_kem.c

```diff
@@ -23,15 +23,17 @@
 #include "utils/s2n_mem.h"
 #include "utils/s2n_safety.h"
 
-const struct s2n_kem s2n_sike_r1_p503 = {
-        .kem_extension_id = SIKEp503r1_KEM,
-        .public_key_length = SIKE_P503_PUBLIC_KEY_BYTES,
-        .private_key_length = SIKE_P503_SECRET_KEY_BYTES,
-        .shared_secret_key_length = SIKE_P503_SHARED_SECRET_BYTES,
-        .ciphertext_length = SIKE_P503_CIPHERTEXT_BYTES,
-        .generate_keypair = &SIKE_P503_crypto_kem_keypair,
-        .encapsulate = &SIKE_P503_crypto_kem_enc,
-        .decapsulate = &SIKE_P503_crypto_kem_dec,
+const struct s2n_kem s2n_sike_supported_params[1] = {
```

Comment:
I am planning on keeping the BIKE and SIKE lists separate everywhere. In `s2n_client_extensions_send` I can more easily decide which to send based on the cipher preferences and in `s2n_set_cipher_and_cert_as_tls_server` go though and decide if the candidate is BIKE and check that there is a mutual BIKE parameter. This PR doesn't have all of that logic because right now with just SIKE its easier to work around this.